mm2src/gui_storage/src/account/storage/sqlite_storage.rs

```diff
@@ -0,0 +1,580 @@
+            SqlError::Utf8Error(_) | SqlError::NulError(_) | SqlError::ToSqlConversionFailure(_) => {
+                AccountStorageError::ErrorSerializing(error)
+            },
+            _ => AccountStorageError::Internal(error),
+        }
+    }
+}
+
+impl AccountId {
+    /// An alternative to [`AccountId::to_tuple`] that returns SQL compatible types.
+    fn to_sql_tuple(&self) -> (i64, Option<i64>, Option<String>) {
+        let (account_type, account_idx, device_pubkey) = self.to_tuple();
+        (
+            account_type as i64,
+            account_idx.map(|idx| idx as i64),
+            device_pubkey.map(|pubkey| pubkey.to_string()),
+        )
+    }
+
+    /// An alternative to [`AccountId::try_from_tuple`] that takes SQL compatible types.
+    pub(crate) fn try_from_sql_tuple(
+        account_type: i64,
+        account_idx: Option<u32>,
+        device_pubkey: Option<String>,
+    ) -> AccountStorageResult<AccountId> {
+        let account_type = AccountType::try_from(account_type)?;
+        let device_pubkey = device_pubkey
+            // Map `Option<String>` into `Option<Result<HwPubkey, _>>`
+            .map(|pubkey| HwPubkey::from_str(&pubkey))
+            // Transpose `Option<Result<HwPubkey, _>>` into `Result<Option<HwPubkey, _>>`
+            .transpose()
+            .map_to_mm(|e| AccountStorageError::ErrorDeserializing(e.to_string()))?;
+        AccountId::try_from_tuple(account_type, account_idx, device_pubkey)
+    }
+}
+
+impl EnabledAccountId {
+    /// An alternative to [`EnabledAccountId::to_pair`] that returns SQL compatible types.
+    fn to_sql_pair(self) -> (i64, Option<i64>) {
+        let (account_type, account_idx) = self.to_pair();
+        (account_type as i64, account_idx.map(|idx| idx as i64))
+    }
+
+    /// An alternative to [`EnabledAccountId::try_from_pair`] that takes SQL compatible types.
+    pub(crate) fn try_from_sql_pair(
+        account_type: i64,
+        account_idx: Option<u32>,
+    ) -> AccountStorageResult<EnabledAccountId> {
+        let account_type = EnabledAccountType::try_from(account_type)?;
+        EnabledAccountId::try_from_pair(account_type, account_idx)
+    }
+}
+
+pub(crate) struct SqliteAccountStorage {
+    conn: SqliteConnShared,
+}
+
+impl SqliteAccountStorage {
+    pub(crate) fn new(ctx: &MmArc) -> AccountStorageResult<SqliteAccountStorage> {
+        let shared = ctx
+            .sqlite_connection
+            .as_option()
+            .or_mm_err(|| AccountStorageError::Internal("'MmCtx::sqlite_connection' is not initialized".to_owned()))?;
+        Ok(SqliteAccountStorage {
+            conn: Arc::clone(shared),
+        })
+    }
+
+    fn lock_conn(&self) -> AccountStorageResult<MutexGuard<Connection>> {
+        self.conn
+            .lock()
+            .map_to_mm(|e| AccountStorageError::Internal(format!("Error locking sqlite connection: {}", e)))
+    }
```

Comment:
`lock_conn` naming can be confusing for some people. SQLite  supports db-level locking so maybe renaming this function to `lock_conn_app_level` or `lock_conn_mutex` can be more cleaner.

Reply:
I like `lock_conn_mutex`, will do, thanks!

Reply:
Done